whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/LuceneAutocompleteSearch.java

```diff
@@ -0,0 +1,179 @@
+package net.ripe.db.whois.api.autocomplete;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import net.ripe.db.whois.api.fulltextsearch.FullTextAnalyzer;
+import net.ripe.db.whois.api.fulltextsearch.FullTextIndex;
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectTemplate;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.facet.taxonomy.TaxonomyReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.TopFieldDocs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Component
+@Conditional(LuceneSearchCondition.class)
```

Comment:
Could you use ConditionalOnProperty instead? Then you can remove the Conditional classes. Define as the opposite to the ElasticSearch class.

```
@ConditionalOnProperty(prefix = "elasticsearch", name = "enabled", havingValue = "false", matchIfMissing="true")
```

Reply:
Same as before: ConditionalOnProperty is springBoot annotation in spring we have only Conditional